Gems/Atom/RPI/Code/Source/RPI.Public/Shader/ShaderResourceGroup.cpp

```diff
@@ -31,13 +31,22 @@ namespace AZ
             AZ_Assert(srgLayout != nullptr, "Failed to find SRG with name %s, using supervariantIndex %u from shaderAsset %s", srgName.GetCStr(),
                 supervariantIndex.GetIndex(), shaderAsset.GetHint().c_str());
 
-            // Create a uuid that combines string data for both the srgLayout and the srgName
-            AZ::Uuid instanceUuid =
-                AZ::Uuid::CreateData(reinterpret_cast<const AZStd::byte*>(srgLayout->GetUniqueId().data()), srgLayout->GetUniqueId().size())
-                + AZ::Uuid::CreateData(reinterpret_cast<const AZStd::byte*>(srgName.GetStringView().data()), srgName.GetStringView().size());
-
-            // Use the supervariantIndex as the subId for the InstanceId, since it is already an integer
-            return Data::InstanceId(instanceUuid, supervariantIndex.GetIndex());
+            // Create the InstanceId By combining data from the asset ID, asset pointer, super variant index, and the SRG name hash. Using
+            // the asset pointer will help make the instance ID more unique for different versions of the same asset. It is possible to use
+            // a more robust value, like a hash of the data from the asset.
+            struct InstanceIdData
+            {
+                const Data::AssetId m_assetId;
+                const Data::AssetData* m_assetPtr;
```

Comment:
Do we want m_assetPtr here? If you a SRG1 and SRG2 in different files but containing the same layout they should be using the same SRGPool. This is probably true for partial SRGs. Having an asset pointer would mean that every time we have a new SRG it will create a new unique uuid and as a result it will create new SRGPool and allocate a new SRG from it. Basically it will defeat the whole purpose of a SRGPool as every SRG will now get a new pool. I think we should remove m_assetPtr from this struct. m_srgLayoutHash should be good enough.

Reply:
@moudgils wouldn't saving SRG1 and SRG2 in different files _already_ cause them to use different SRG pools, since they'll have different asset IDs?

Reply:
yeah the best approach here is to just use srg name and layout hash. and make sure to test properly.